lua/ulx/modules/sh/cfc_checklua.lua

```diff
@@ -2,40 +2,114 @@ CFCUlxCommands.checklua = CFCUlxCommands.checklua or {}
+        local any = false
+        local badConvars = {}
+
+        for _, convar in pairs( CONVARS_TO_CHECK ) do
+            local convarValue = net.ReadBool()
+
+            if convarValue then
+                any = true
+                badCovnars[convar] = convarValue
+                ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's " .. convar .. "value is " .. tostring( convarValue ), ply )
+            end
+        end
+
+        return any, badConvars
+    end
+
+    local function unawaitTimerName( steamId )
+        return "CFC_ULX_AutoStatsUnAwait_" .. steamId
+    end
+
+    net.Receive( "CFC_ULX_StatCheck", function( _, ply )
+        if not awaitingResponse[ply] then return end
+        awaitingResponse[ply] = nil
+
+        receiveConvars( ply )
+    end )
 
-local awaitingResponse = {}
+    net.Receive( "CFC_ULX_AutoStatCheck", function( _, ply )
+        if not awaitingResponse[ply] then return end
+        awaitingResponse[ply] = nil
+        timer.Remove( unawaitTimerName( ply:GetSteamID() ) )
 
-net.Receive( "CFC_ULX_StatCheckSV", function( _, ply )
-    if not awaitingResponse[ply] then return end
-    local convar = net.ReadBool()
-    ulx.fancyLogAdmin( awaitingResponse[ply], true, "#T's sv_allowcslua value is " .. tostring( convar ), ply )
+        local anyConvars, badConvars = receiveConvars( ply )
 
-    awaitingResponse[ply] = nil
-end )
+        if anyConvars then
+            print( "Banning " .. tostring( ply ) .. " for having bad convars." )
+            PrintTable( badConvars )
 
-function cmd.checkluaPlayers( callingPlayer, targetPlayers )
-    for _, ply in pairs( targetPlayers ) do
-        awaitingResponse[ply] = callingPlayer
-        net.Start( "CFC_ULX_StatCheckCL" )
-        net.Send( ply )
+            ulx.ban( nil, ply, AUTO_BAN_LENGTH, AUTO_BAN_REASON )
+            sendAwarn( ply )
+        end
+    end )
+
+    function cmd.checkluaPlayers( callingPlayer, targetPlayers )
+        for _, ply in pairs( targetPlayers ) do
+            awaitingResponse[ply] = callingPlayer
+            net.Start( "CFC_ULX_StatCheckCL" )
+            net.Send( ply )
+        end
     end
+
+    gameevent.Listen( "player_connect" )
+    hook.Add( "player_connect", "CFC_ULX_AutoStatsCheckOnConnect", function( data )
+        local steamId = data.networkid
+
+        local ply = player.GetBySteamID( steamId )
+        awaitingResponse[ply] = true
+
+        timer.Create( unawaitTimerName( steamId ), 60 * 15, 1, function()
+            awaitingResponse[ply] = nil
+        end )
+    end )
 end
 
 if CLIENT then
-    net.Receive( "CFC_ULX_StatCheckCL", function()
-        local convarBool = GetConVar( "sv_allowcslua" ):GetBool()
-        net.Start( "CFC_ULX_StatCheckSV" )
-        net.WriteBool( convarBool )
+    local function respondToQuery( networkString )
+        net.Start( networkString )
+
+        for _, convar in pairs( CONVARS_TO_CHECK ) do
+            net.WriteBool( GetConVar( convar ):GetBool() )
```

Comment:
Most cheats create their own convars, GetConVar will error when the convar doesn't exist.

Reply:
Docs say GetConvar will return nil if it doesnt exist, but ye that getbool will err